### PR TITLE
Adding CreateRoute to Routes page

### PR DESCRIPTION
**Pull request by AlexRutledge1**
# Describe your changes
**This PR can't be merged until xplat merges the RouteColor PR**
Wrapped the CreateRoute component in a nested Popup so that the user is prompted with a confirmation message when creating a route. 
Tested this component and it successfully creates a draft route as long as the user fills the required fields.
Since it was a test run I kept the route as a draft so it could be deleted later.
image
image

# Issue ticket number and link
#24

### CODE REVIEW

File: src/components/Route/CreateRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  if (type === RouteType.Competition)
+    return getAllCompetitionRouteClassifiers();
+  if (type === RouteType.Leadclimb)
+    return getAllLeadclimbRouteClassifiers();
+  return getAllTopropeRouteClassifiers();
+};
+
+const Ropes = [
+  '1', '2', '3', '4', '5', '6', '7', '8', '9'
+];
```
Comment (DanielW48) [inline src/components/Route/CreateRoute.tsx:28]: Should this be in xplat? Lmao

Comment (JacobSteinebronn) [reply src/components/Route/CreateRoute.tsx:28]: should this be ~~in xplat~~?

Comment (AlexRutledge1) [reply src/components/Route/CreateRoute.tsx:28]: Maybe? If mobile will use it for any kind of drop-down then yeah but I think they'll just be seeing ropes when fetching a route.

File: src/components/Route/CreateRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    // check that no active route has the same name
+    createRoute(
+      {
+        name: name, 
+        classifier: new RouteClassifier(rawgrade!, type!), 
+        setter: setter,
+        setterRawName: overrideSetterString,
+        description: description,
+        rope: rope,
+        tech: rules,
```
Comment (DanielW48) [inline src/components/Route/CreateRoute.tsx]: this should be using the newest version of xplat with naturalRules field, but this will be resolved once you grab the newest version for RouteColor enum anyway 👍

File: src/components/Route/CreateRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  return (
+    <Popup open={isOpen} onClose={() => setIsOpen(false)} closeOnDocumentClick={false} nested modal>
+      <div>
+        <Popup nested modal open={confirmationBoxOpen} onClose={() => setConfirmationBoxOpen(false)}>
+          <VStack alignItems='center'>
+            <Text variant='header'>
+              Are you sure?
+            </Text>
+            <Text variant='body'>
+              Are you sure you want to create this route? The name and grade must be unique and cannot be changed later.
```
Comment (DanielW48) [inline src/components/Route/CreateRoute.tsx:106]: so this is pretty good, are we planning to make the routes editable besides name/grade? I could back that

File: src/components/Route/CreateRoute.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
+import { Button, FormControl, Input, Text, VStack, Select, HStack } from 'native-base';
+import { useState } from 'react';
+import { createRoute } from '../../xplat/api';
+import Popup from 'reactjs-popup';
+import 'reactjs-popup/dist/index.css';
+import { User } from '../../xplat/types/user';
+import { RouteType, RouteClassifier, RouteColor, RouteTech } from '../../xplat/types/route';
```
Comment (DanielW48) [inline src/components/Route/CreateRoute.tsx]: wait, RouteColor enum does not exist in this branch's version of xplat. Or perhaps im mistaken? shouldnt this be included in another commit where u merge new xplat?

File: src/components/Route/CreateRoute.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            {Object.values(RouteTech).map( (rules) => {
+              return <Select.Item key={rules} label={rules} value={rules}/>;
+            })}
+          </Select>
+          <FormControl.Label>Setter</FormControl.Label>
+          {overrideSetterBool ?
+            <Input isRequired={false} type='text' onChangeText={setOverrideSetterString}
+              placeholder='Setter (optional)'/>
+            :
+            <Text>This will be a user search component</Text>
```
Comment (DanielW48) [inline src/components/Route/CreateRoute.tsx:177]: bet, TODO/extra issue for putting this into this component when UserSearch is ready?

File: src/components/Route/CreateRoute.tsx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        setter: setter,
+        setterRawName: overrideSetterString,
+        description: description,
+        rope: rope,
+        tech: rules,
+        thumbnail: thumbnailFile,
+        color: routeColor,
+      })
+      .then( (route) => {
+        // handle successful create route (refresh Routes page or pass route to parent component)
```
Comment (DanielW48) [inline src/components/Route/CreateRoute.tsx:89]: I think we should also immediately upgrade its status to active route. When we allow draft routes im thinkin we make a 'Save as Draft' button in addition to the 'Create Route' button on this popup

File: src/components/Route/CreateRoute.tsx (hunk 4 above)
Comment (AlexRutledge1) [reply src/components/Route/CreateRoute.tsx]: I didn't update xplat before adding in RouteColor. but I just updated it in the latest commit so this is no longer inconsistent.

File: src/components/Route/CreateRoute.tsx (hunk 5 above)
Comment (AlexRutledge1) [reply src/components/Route/CreateRoute.tsx:177]: Just made the issue

**Review by DanielW48 — APPROVED** [review]
Sweeeet. You can mark the epic completed as well 🙌 good stuff man.
